core/javax.cache/src/main/java/org/wso2/carbon/caching/impl/CacheManagerFactoryImpl.java

```diff
@@ -148,6 +148,20 @@ public void removeCacheManager(CarbonCacheManager cacheManager, String tenantDom
         }
     }
 
+    /**
+     * Remove all the Caches and CacheManagers of the specified tenant
+     *
+     * @param tenantDomain The domain of the tenant whose caches need to be removed
+     */
+    public void removeAllCacheManagers(String tenantDomain) {
+        Map<String, CacheManager> cacheManagers = globalCacheManagerMap.get(tenantDomain);
+        for (CacheManager cacheManager : cacheManagers.values()) {
```

Comment:
Potential NPE, check for null "cacheManagers"

Reply:
fixed